### ISSUE TITLE
Crime and Criminal Investigation 

### ISSUE THREAD

**Issue by shirishag75** [mars-sim/mars-sim#91]
Mostly in popular culture, the deep space exploration era or slightly less ambitious having a settlement in Mars would be a dream come true for the country/countries/corporations who would want to have it and the bio-terrorists, terrorists, foreign governments who might do anything to scuttle the plans/works.

**Comment by mokun** [comment]
> a few stories which had been looping in my head now for quite sometime

so true about having a story to tell. 

Life would have been very mechanical and boring flat out if there's no story to tell.

We have a new job called "Reporter". 

Supposedly, his job is documenting and highlighting the important things he was able to capture. 

I don't know how to programmatically make that happen. 

We have in place a list of Malfunctions that can happen any time. 

At some points, we'll "connect" how a malfunction happens and who may have risked his life by coming to the scene to rescue the wounded, etc.  That would be an epic story to tell !

To really "write" a story, we will need a language parser.

**Comment by mokun** [comment]
See [Malfunctions](https://github.com/mars-sim/mars-sim/wiki/Malfunctions-and-Field-Reliability) wiki page

**Comment by mokun** [comment]
> After a series of accidents or death (natural or otherwise) s(he) is coerced by the Corporation/State to 

See [Malfunctions](https://github.com/mars-sim/mars-sim/wiki/Malfunctions-and-Field-Reliability) wiki page.

Each malfunction comes with a scope. e.g. A `Class A Fire` will only happen to the `Waste Disposal` and `Life Support` System, while a `Vehicle Fire` will only happen to a vehicle, etc.

When a malfunction or accident happens, the Historical Event Manager will record what happen.

**Comment by Urwumpe** [comment]
Maybe we can extend the death reports that way by creating a special science project to investigate the death?

**Comment by shirishag75** [comment]
Death reports should be part of permanent records, once record of death entered in, details such as date, place, time and cause of death should be permanently stored to be called up in an instant's notice. How we store those records is upto you guys. Maybe it would be part of the HistoricalEventManager fully or partly in 'HistoricalEventManager' and partly in 'Malfunctions' .  This I will let it be in both your capable hands :)

**Comment by mokun** [comment]
> Death reports should be part of permanent records, once record of death entered in, details such as date, place, time and cause of death should be permanently stored to be called up in an instant's notice. 

Yes agree !

**Comment by mokun** [comment]
I'm putting in more codes such as waiting for one whose job is Doctor to perform postmortem exam. No "proper burial" or funeral ceremony just yet. 

In future, we may code it in such a way that the cause of death is controversial. 

When you say criminal investigation, it implies murder or suicide-- something or known, be it impulsive, planned or premeditated.

Currently, it's somewhat direct. Whatever action calls setDead() in PhysicalCondition class is the cause.

The murky part is the person's medical condition. Sometimes a person has untreated illness and is lingering around and is degrading his/her health.

When one dies, his/her health problem will be reported alongside as a reference. But it can easily be the main contributor to the cause of illness. 

What do we know about life, let alone how it's supposed to end ? Is it just the heart stop beating and/or the brain wave ceases. Many people in fact bounce back from being dead after hours and days and were well documented :D

**Comment by Urwumpe** [comment]
Well, from what I can tell, death is not the end of life, your body just stops working as one unit, instead the organs and cells will try to keep going to the point of you digesting yourself. That is what makes death so messy. 

I think the better term for what is needed on Mars would be a "Accident Investigation Board", after all, it is still not implemented that people simply die of old age (aka tear and wear) in MSP yet. Of course, later, people COULD die of natural causes. But then, it is very natural to die without oxygen.

But getting back to AIB, instead of discussing about where life ends and death starts, we could maybe focus on a definition when an accident happened? Maybe that is simpler to define. For example, a mild accident could already happen, when a task or process fails to meet its post-conditions. Then we would just need a way to define the post-conditions, but maintaining the health of the settler performing the task seems to be a good general post-condition. For most tasks, this could be extended to the object as well: Driving a rover should not result in a damaged rover. Of course, there would then need to be another factor needed to avoid AIBs about lacerations. 

What about using "Caused permanent or long-term damage to person, settlement or vehicle" as a way to classify it? Of course, there is little less permanent than death, but this way, already surviving with a lost finger would justify an investigation. An accident resulting just in a life-support system being out of order for weeks on the other hand has a big effect on the lifes of the settlers, so of course it makes sense to consider it important enough.

Maybe the size of an investigation could then be defined by the magnitude of the permanent damage. A lost finger is bad, but does not justify 40 people sitting around in a conference. For such small damages, one investigator should be enough maybe, even the person which lost the finger or the supervisor responsible for him/her.

**Comment by mokun** [comment]
> the better term for what is needed on Mars would be a "Accident Investigation Board",

Yes. AIB is a cool concept. Here in the U.S., we have NTSB that investigates airplane related incidents. You can say that the concept of accident investigation was being discussed here and there in the past :D but we never get the motivation to move forward coding it.

> Maybe the size of an investigation could then be defined by the magnitude of the permanent damage.

It's true that they will need to sort out what worth an investigation. As things get more standardized in the settlement, it makes sense they need it.

I believe we are now in a better shape than ever to code toward assembling a group of people as an AIB for serious incidents such as someone who die, like you said.

Let's see how we can do it, will it be in the form of a new `Task` that is collaborative in nature ?

**Comment by mokun** [comment]
Sure ! One thing we'll need to make sure is that players can gain insights into every step of way if they want to, which mean we do need an interface to nicely present the findings, and may be even asking them to participate in the process of agreeing or not agreeing, since they are the commander or the Earth Mission Control :D
